extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -254,19 +253,31 @@ function createDicomWebApi(dicomWebConfig, userAuthenticationService) {
       madeInClient
     ) => {
       const enableStudyLazyLoad = false;
-      wadoDicomWebClient.headers = generateWadoHeader();
-      // data is all SOPInstanceUIDs
-      const data = await retrieveStudyMetadata(
-        wadoDicomWebClient,
-        StudyInstanceUID,
-        enableStudyLazyLoad,
-        filters,
-        sortCriteria,
-        sortFunction
-      );
-
-      // first naturalize the data
-      const naturalizedInstancesMetadata = data.map(naturalizeDataset);
+      clientManagerObj.generateWadoHeader();
```

Comment:
`generateWadoHeader` looks like it takes a `config` argument yet one is NOT being passed here. That said, consider just having a `clientManager.generateWadoHeader` method instead of a `generateWadoHeader` method and a `setWadoHeaders` method.